### DISCUSSION TITLE
Polar vs nonpolar molecules

### DISCUSSION THREAD

**Discussion by edgarsmdn** [rdkit/rdkit#5331] [Q&A]
Hello, is there a convenient way in the RDKit to get whether a given molecule is **polar** or **nonpolar**? Thanks!

**Comment by greglandrum** [comment]
This is going to depend on how you define the terms "polar" and "nonpolar"; there's not one universal definition.

Do you have a particular definition which you want to use?

**Comment by edgarsmdn** [comment]
I do not have a precise definition actually (maybe due to this gray area in my understanding between the actual border between polar vs. nonpolar molecules (specially when dealing with large ones)). But, I guess I am looking for a way to detect the (general) symmetrical electrical charge distribution for nonpolar vs. the uneven distribution of charge in polar small molecules.

**Reply by UnixJunkie** [reply]
Maybe, prepare several molecules in 3D w/ partial charges (some known hydrophobic ones and some known hydrophilic ones). Then, compute the dipole moment for those molecules. Probably, the magnitude of the dipole moment could allow you to classify molecules between polar Vs. non polar.
Maybe, you will need to consider several conformers per molecule to obtain a robust method.
Maybe, some charge models will work better than others.
